### PR TITLE
Rebase on llvm-13, small CI improvements and drop support for old python

### PR DESCRIPTION
**Pull request by kamino**

### CODE REVIEW

**Comment by kamino** [comment]
@aguinet any review or comment to improve this pull request is welcome 😇

File: .github/workflows/dist_publish.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
         mkdir llvm/install
         cd llvm/build
         cmake ../llvm -DLLVM_BUILD_EXAMPLES=OFF -DCLANG_ENABLE_ARCMT=OFF -DCLANG_ENABLE_STATIC_ANALYZER=OFF -DLLVM_BUILD_TOOLS=ON -DCLANG_BUILD_TOOLS=OFF -DLLVM_INSTALL_UTILS=ON -DLLVM_TARGETS_TO_BUILD=X86 -DLLVM_ENABLE_PROJECTS=clang -DLLVM_ENABLE_BINDINGS=OFF -DCMAKE_C_COMPILER=cl -DCMAKE_CXX_COMPILER=cl -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX="$PWD/../install" -DLLVM_ENABLE_THREADS=OFF -DLLVM_ENABLE_ZLIB=OFF  -G Ninja
-        ninja install
+        ninja -j1 install
```
Comment (aguinet) [inline .github/workflows/dist_publish.yml:43]: Why `-j1` here?

File: examples/CMakeLists.txt [hunk 2]
```diff
@@ -9,10 +9,14 @@ if (BUILD_TESTS)
 
     set(PYDFFI_LIB_DIR $<TARGET_FILE_DIR:pydffi>)
     configure_file("lit.site.cfg.in" "${CMAKE_CURRENT_BINARY_DIR}/lit.site.cfg" ESCAPE_QUOTES @ONLY)
+
+    # NOTE: added --max-time to timeout lit after specified seconds.
+    # Default timeout is one week
     add_custom_target(check_python_examples
-      COMMAND ${CMAKE_COMMAND} -E env PYDFFI_DIR=${PYDFFI_LIB_DIR} "${PYTHON_EXECUTABLE}" "${LIT_RUNNER}" "${CMAKE_CURRENT_BINARY_DIR}" -v
+      COMMAND ${CMAKE_COMMAND} -E env PYDFFI_DIR=${PYDFFI_LIB_DIR} "${PYTHON_EXECUTABLE}" "${LIT_RUNNER}" "${CMAKE_CURRENT_BINARY_DIR}" -v --max-time=900
```
Comment (aguinet) [inline examples/CMakeLists.txt:16]: Why this `max-time`?

**Comment by aguinet** [comment]
Superseeded by #75

**Pull request closed without merging**